File: app/src/main/java/com/roman/gurdan/sudo/pro/fragment/DataFragment.java
package com.roman.gurdan.sudo.pro.fragment;

import android.os.Bundle;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.roman.gurdan.sudo.pro.App;
import com.roman.gurdan.sudo.pro.R;
import com.roman.gurdan.sudo.pro.base.BaseFragment;
import com.roman.gurdan.sudo.pro.data.db.GameData;
import com.roman.gurdan.sudo.pro.view.DataView;

import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers;
import io.reactivex.rxjava3.core.Observable;
import io.reactivex.rxjava3.disposables.CompositeDisposable;
import io.reactivex.rxjava3.functions.Consumer;
import io.reactivex.rxjava3.functions.Function;
import io.reactivex.rxjava3.schedulers.Schedulers;

public class DataFragment extends BaseFragment {

    private DataView total;
    private DataView totalWin;

    private CompositeDisposable disposables = new CompositeDisposable();

    @Override
    protected int getLayoutId() {
        return R.layout.fragment_data;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        total = view.findViewById(R.id.total);
        totalWin = view.findViewById(R.id.totalWin);
        setup();
    }

    @Override
    public void onDestroyView() {
        if (disposables != null && !disposables.isDisposed()) {
            disposables.dispose();
        }
        super.onDestroyView();
    }

    private void setup() {
        disposables.add(Observable.just(1)
                .map(new Function<Integer, Integer>() {
                    @Override
                    public Integer apply(Integer integer) throws Throwable {
                        return GameData.of(App.instance)
                                .gameDao().getCount();
                    }
                })
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(new Consumer<Integer>() {
                    @Override
                    public void accept(Integer integer) throws Throwable {
                        if (total != null) total.setMsg(String.valueOf(integer));
                    }
                }));
        disposables.add(Observable.just(1)
                .map(new Function<Integer, Integer>() {
                    @Override
                    public Integer apply(Integer integer) throws Throwable {
                        return GameData.of(App.instance)
                                .gameDao().getGamesResultCount(1);
                    }
                })
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(new Consumer<Integer>() {
                    @Override
                    public void accept(Integer integer) throws Throwable {
                        if (totalWin != null) totalWin.setMsg(String.valueOf(integer));
                    }
                }));
    }

}

File: app/src/main/java/com/roman/gurdan/sudo/pro/fragment/DataFragment.kt
package com.roman.gurdan.sudo.pro.fragment

import android.os.Bundle
import android.view.View
import android.view.ViewStub
import com.roman.garden.core.Easy
import com.roman.gurdan.sudo.pro.R
import com.roman.gurdan.sudo.pro.base.BaseFragment
import com.roman.gurdan.sudo.pro.data.db.GameData
import com.roman.gurdan.sudo.pro.view.DataView
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.MainScope
import kotlinx.coroutines.flow.flow
import kotlinx.coroutines.flow.flowOn
import kotlinx.coroutines.launch

class DataFragment : BaseFragment() {

    private lateinit var total: DataView
    private lateinit var totalWin: DataView
    private lateinit var stubChart: ViewStub
    private lateinit var stubEmpty: ViewStub

    override fun getLayoutId(): Int = R.layout.fragment_data

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        total = view.findViewById(R.id.total)
        totalWin = view.findViewById(R.id.totalWin)
        stubChart = view.findViewById(R.id.stubChart)
        stubEmpty = view.findViewById(R.id.stubEmpty)
        setup()

    }

    private fun setup() {
        MainScope().launch {
            flow<Int> {
                val count = GameData.instance.gameDao().getCount()
                emit(count)
            }
                .flowOn(Dispatchers.IO)
                .collect { i ->
                    try {
                        total.setMsg("$i")
                        if (i > 0) stubChart.inflate()
                        else stubEmpty.inflate()
                    } catch (e: Exception) {
                    }
                }
        }
        MainScope().launch {
            flow<Int> {
                val count = GameData.instance.gameDao().getGamesResultCount(1)
                emit(count)
            }.flowOn(Dispatchers.IO)
                .collect { i ->
                    try {
                        totalWin.setMsg("$i")
                    } catch (e: Exception) {
                    }
                }
        }
    }

}